### PR TITLE
fix: add DB migration to add workspace

### PR DESCRIPTION
**Pull request by pengx17**
This pr allows the user to add a old db file without sub docs. When loading an old db, it will check if it had doc_id column. If not, it will firstly clone the source db to a tmp folder, migrate it, persist to db file and then use the new db file path as the new db source.

### CODE REVIEW

File: apps/electron/src/helper/db/migration.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  const rows = await db.getAllUpdates();
+  const originalDoc = new Y.Doc();
+
+  // 1. apply all updates to the root doc
+  rows.forEach(row => {
+    Y.applyUpdate(originalDoc, row.data);
+  });
+
+  console.log(originalDoc.toJSON());
```
Comment (himself65) [inline apps/electron/src/helper/db/migration.ts]: No console

File: apps/electron/tsconfig.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
     "outDir": "dist",
     "moduleResolution": "node",
     "resolveJsonModule": true,
-    "noImplicitOverride": true
+    "noImplicitOverride": true,
+    "verbatimModuleSyntax": false
```
Comment (himself65) [inline apps/electron/tsconfig.json]: Why

File: packages/native/index.d.ts [hunk 3]
```diff
@@ -38,6 +38,12 @@ export interface InsertRow {
   docId?: string;
   data: Uint8Array;
 }
+export const enum ValidationResult {
```
Comment (himself65) [inline packages/native/index.d.ts]: Remove const here could helpful for removing compiler options

**Review by himself65 — CHANGES_REQUESTED** [review]

File: packages/native/index.d.ts (hunk 3 above)
Comment (pengx17) [reply packages/native/index.d.ts:41]: This is auto generated. Not sure if there is a better option 😳

File: apps/electron/tsconfig.json (hunk 2 above)
Comment (pengx17) [reply apps/electron/tsconfig.json:15]: verbatimModuleSyntax cannot be used with `export const` from an external module.

File: packages/native/index.d.ts (hunk 3 above)
Comment (himself65) [reply packages/native/index.d.ts]: Related: https://github.com/napi-rs/napi-rs/issues/1345

you need `--const-enum false`

**Review by himself65 — CHANGES_REQUESTED** [review]

**Comment by pengx17** [comment]
> Related: https://github.com/napi-rs/napi-rs/issues/1345
> you need --const-enum false

Looks like the v2 version we are using not honoring ts.constEnum option in package.json. I proposed a PR upstream: https://github.com/napi-rs/napi-rs/pull/1647

**Review by himself65 — APPROVED** [review]

**Pull request merged**